compiler/rustc_middle/src/ty/generic_args.rs

```diff
@@ -440,7 +440,7 @@ impl<'tcx> GenericArgs<'tcx> {
         target_args: GenericArgsRef<'tcx>,
     ) -> GenericArgsRef<'tcx> {
         let defs = tcx.generics_of(source_ancestor);
-        tcx.mk_args_from_iter(target_args.iter().chain(self.iter().skip(defs.params.len())))
+        tcx.mk_args_from_iter(target_args.iter().chain(self.iter().skip(defs.count())))
```

Comment:
This is a drive-by -- not needed currently, per se, but it is a bug if you rebase from anything that has parent generics (i.e. rebasing RPIT generics that are stacked on top of a trait method signature onto a impl method signature).